src/parser/vp9_parser.cpp

```diff
@@ -73,30 +79,117 @@ rocDecStatus Vp9VideoParser::ParseVideoData(RocdecSourceDataPacket *p_data) {
 
 ParserResult Vp9VideoParser::ParsePictureData(const uint8_t *p_stream, uint32_t pic_data_size) {
     ParserResult ret = PARSER_OK;
-    printf("Frame %d: pic_data_size = %d.................\n", pic_count_, pic_data_size); // Jefftest
-    if ((ret = ParseUncompressedHeader(const_cast<uint8_t*>(p_stream), pic_data_size)) != PARSER_OK) {
-        return ret;
-    }
-    // Init Roc decoder for the first time or reconfigure the existing decoder
-    if (new_seq_activated_) {
-        if ((ret = NotifyNewSequence(&uncompressed_header_)) != PARSER_OK) {
+
+    CheckSuperframe(p_stream, pic_data_size);
+
+    uint8_t *pic_data_ptr = const_cast<uint8_t*>(p_stream);
+    for (int frame_index = 0; frame_index < num_frames_in_chunck_; frame_index++) {
+        if ((ret = ParseUncompressedHeader(pic_data_ptr, frame_sizes_[frame_index])) != PARSER_OK) {
             return ret;
         }
-        new_seq_activated_ = false;
-    }
-    pic_stream_data_ptr_ = const_cast<uint8_t*>(p_stream);
-    pic_stream_data_size_ = pic_data_size;
-    num_slices_ = 1;
+        // Init Roc decoder for the first time or reconfigure the existing decoder
+        if (new_seq_activated_) {
+            if ((ret = NotifyNewSequence(&uncompressed_header_)) != PARSER_OK) {
+                return ret;
+            }
+            new_seq_activated_ = false;
+        }
 
-    // Decode the picture
-    if ((ret = SendPicForDecode()) != PARSER_OK) {
-        ERR(STR("Failed to decode!"));
-        return ret;
+        if (uncompressed_header_.show_existing_frame) {
+            int disp_idx = dpb_buffer_.virtual_buffer_index[uncompressed_header_.frame_to_show_map_idx];
+            if (disp_idx == INVALID_INDEX) {
+                ERR("Invalid existing frame index to show.");
+                return PARSER_INVALID_ARG;
+            }
+            if (pfn_display_picture_cb_) {
+                disp_idx = dpb_buffer_.frame_store[disp_idx].dec_buf_idx;
+                decode_buffer_pool_[disp_idx].use_status |= kFrameUsedForDisplay;
+                decode_buffer_pool_[disp_idx].pts = curr_pts_;
+                // Insert into output/display picture list
+                if (num_output_pics_ >= dec_buf_pool_size_) {
```

Comment:
Suggest reversing the condition as else part is more likely to execute

Reply:
Done.